libs/vkd3d/command.c

```diff
@@ -1681,6 +1681,284 @@ static void d3d12_command_list_invalidate_current_pipeline(struct d3d12_command_
+    VkRenderPassCreateInfo pass_info;
+    VkFramebuffer vk_framebuffer;
+    VkRenderPass vk_render_pass;
+    VkPipelineStageFlags stages;
+    VkAccessFlags access;
+    VkExtent3D extent;
+    VkResult vr;
+
+    attachment_desc.flags = 0;
+    attachment_desc.format = view->format->vk_format;
+    attachment_desc.samples = vk_samples_from_dxgi_sample_desc(&resource->desc.SampleDesc);
+    attachment_desc.loadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+    attachment_desc.storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+    attachment_desc.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+    attachment_desc.stencilStoreOp = VK_ATTACHMENT_STORE_OP_STORE;
+    attachment_desc.initialLayout = resource->common_layout;
+    attachment_desc.finalLayout = resource->common_layout;
+
+    attachment_ref.attachment = 0;
+    attachment_ref.layout = view->info.texture.vk_layout;
+
+    subpass_desc.flags = 0;
+    subpass_desc.pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;
+    subpass_desc.inputAttachmentCount = 0;
+    subpass_desc.pInputAttachments = NULL;
+    subpass_desc.colorAttachmentCount = 0;
+    subpass_desc.pColorAttachments = NULL;
+    subpass_desc.pResolveAttachments = NULL;
+    subpass_desc.pDepthStencilAttachment = NULL;
+    subpass_desc.preserveAttachmentCount = 0;
+    subpass_desc.pPreserveAttachments = NULL;
+
+    if (!rect_count)
+    {
+        if (clear_aspects & (VK_IMAGE_ASPECT_COLOR_BIT | VK_IMAGE_ASPECT_DEPTH_BIT))
+            attachment_desc.loadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
+
+        if (clear_aspects & (VK_IMAGE_ASPECT_STENCIL_BIT))
+            attachment_desc.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
+
+        /* Ignore 3D images as re-initializing those may cause us to
+         * discard the entire image, not just the layers to clear. */
+        if (clear_aspects == view->format->vk_aspect_mask &&
+                resource->desc.Dimension != D3D12_RESOURCE_DIMENSION_TEXTURE3D)
+            attachment_desc.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+    }
+
+    if (clear_aspects & (VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT))
+    {
+        stages = VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT;
+        access = VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_READ_BIT | VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT;
+        subpass_desc.pDepthStencilAttachment = &attachment_ref;
+    }
+    else
+    {
+        stages = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+        access = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+        subpass_desc.colorAttachmentCount = 1;
+        subpass_desc.pColorAttachments = &attachment_ref;
+    }
+
+    dependencies[0].srcSubpass = VK_SUBPASS_EXTERNAL;
+    dependencies[0].dstSubpass = 0;
+    dependencies[0].srcStageMask = stages;
+    dependencies[0].dstStageMask = stages;
+    dependencies[0].srcAccessMask = 0;
+    dependencies[0].dstAccessMask = access;
+    dependencies[0].dependencyFlags = 0;
```

Comment:
BY_REGION if initialLayout does not imply transition.

Reply:
Actually, no. initialLayout is common layout, nvm.

Reply:
we're pretty much always transitioning from either `common_layout` or `UNDEFINED` to `ATTACHMENT_OPTIMAL` though.

Reply:
yup, so, BY_REGION is indeed not possible.